pkg/virt-launcher/notify-client/client.go

```diff
@@ -231,6 +231,8 @@ func eventCallback(c cli.Connection, domain *api.Domain, libvirtEvent libvirtEve
 	} else {
 		defer d.Free()
 
+		prevStatus := domain.Status.DeepCopy() // Remember current status before it will be changed.
```

Comment:
Maybe just keep the relevant fields around (Status.Status & Status.Reason) instead of reallocating the whole Status object including interface status etc. (assuming DeepCopy does that).

Reply:
Changed.